server/src/main/java/io/crate/replication/logical/seqno/RetentionLeaseHelper.java

```diff
@@ -43,8 +45,12 @@ private static String retentionLeaseSource(String subscriberClusterName) {
         return "logical_replication:" + subscriberClusterName;
     }
 
-    private static String retentionLeaseIdForShard(String subscriberClusterName, ShardId shardId) {
+    @VisibleForTesting
+    static String retentionLeaseIdForShard(String subscriberClusterName, Version remoteVersion, ShardId shardId) {
         var retentionLeaseSource = retentionLeaseSource(subscriberClusterName);
+        if (remoteVersion.before(Version.V_6_1_0)) {
+            return retentionLeaseSource + ":" + "[" + shardId.getIndex().getName() + "][" + shardId.id() + "]";
+        }
```

Comment:
I find this bug confusing to understand. Would this fix cover a scenario where i.e. `5.10` cluster persisted `RetentionLeses.leases` with older key format and upgraded to `6.1` afterwards?

I think in general we need to worry about the persisted retention lease keys.

Reply:
After reading into this more, I think the only safe way to cover all, persistent releases recovered from local fs, mixed cluster and logical replication across different versions is to use the old shardId.toString() pattern for any lease.
Afaik the lease id is just any unique id inside a shard scope, so it shouldn't matter to use a maybe stale indexName and no indexUUID. wdyt?

Reply:
I've figured out that the related `RetentionLeaseHelper.retentionLeaseIdForShard` is only used by the logical replication. Peer recovery leases are using different id's not using the `ShardId.toString()`.

That this helper is using the `ShardId.toString()` may a bad idea from the beginning, using e.g. the LR replicating indexUUID would be a better idea as the lease id should rather express who/what created this lease.
Anyhow, we cannot change this without breaking ongoing logical replications between mixed version clusters, so let's keep the format as it was before.